File: CN_Programs/Practical_Three.cpp
// #include <iostream>
// #include <string>

// using namespace std;

// int main()
// {
//     string Msg = "SSBTCOETJALGAON";
//     char Key = '$';
//     string CTxt = "";
//     string DTxt = "";

//     // Encryption
//     for (int i = 0; i < Msg.length(); i++)
//     {
//         int xor_val = Msg[i] ^ Key;
//         char temp = static_cast<char>(xor_val);
//         CTxt += temp;
//     }

//     cout << "Encryption and Decryption using XOR Cryptography" << endl;
//     cout << "Original Message is: " << Msg << endl;
//     cout << "Encrypted Message is: " << CTxt << endl;

//     // Decryption
//     for (int i = 0; i < CTxt.length(); i++)
//     {
//         int xor1_val = CTxt[i] ^ Key;
//         char temp1 = static_cast<char>(xor1_val);
//         DTxt += temp1;
//     }

//     cout << "Decrypted Message is: " << DTxt << endl;

//     return 0;
// }


// #include<bits/stdc++.h>
// using namespace std;
// int  main()
// {
//     string msg="SSBTCOETJALGAON";
//     char key='$';
//     string CTxt="";
//     string DTxt="";

//     for(int i=0;i<msg.length();i++)
//     {
//         int xor_val=key^msg[i];
//         char temp = static_cast<char>(xor_val);
//         CTxt+=temp;
//     }
//     cout<<"The Original string is: "<<msg<<endl;
//     cout<<"The Encrypted string is: "<<CTxt<<endl;

//     for(int i=0;i<CTxt.length();i++)
//     {
//         int xor_val=key^CTxt[i];
//         char temp = static_cast<char>(xor_val);
//         DTxt+=temp;
//     }
//     cout<<"The Decrypted string is: "<<DTxt<<endl;
//     return 0;
// }


#include<bits/stdc++.h>
using namespace std;
int main()
{
    string msg="ASHVINSUDHIRBARI";
    char key='$';
    string CTxt="";
    string DTxt="";
    for(int i=0;i<msg.length();i++)
    {
        int xor_val=key^msg[i];
        char temp=static_cast<char>(xor_val);
        CTxt+=temp;
    }
    cout<<"The Original Message is: "<<msg<<endl;
    cout<<"The Encrypted Message is: "<<CTxt<<endl;
    for(int i=0;i<CTxt.length();i++)
    {
        int xor_val=key^CTxt[i];
        char temp=static_cast<char>(xor_val);
        DTxt+=temp;
    }
    cout<<"The Decryption Message is: "<<DTxt<<endl;
    return 0;
}

File: CN_Programs/Practical_three.java
//Program : Implementation of XOR Symmetric Cryptographic Algorithm
import java.io.*;
import java.lang.String;
public class Xor4
{
public static voidmain(String args[])
{
String Msg = "SSBTCOETJALGAON";
char Key='$';
String CTxt="";
String DTxt="";
int xor,xor1;
chartemp,temp1;
for(int i=0;i<Msg.length();i++ )
{
xor=Msg.charAt(i) ^ Key;
temp =(char)xor ;
CTxt=CTxt + temp;
}
System.out.println("Encryption and Decryption using XOR Cryptogrpahy");
System.out.println("Original Message is:" +Msg);
System.out.println("Encrypted Message is:" +CTxt);
for(int i=0;i<CTxt.length();i++)
{
xor1=CTxt.charAt(i)^ Key;
temp1 =(char) xor1;
DTxt=DTxt+ temp1;
}
System.out.println("Decrypted Message is:"+DTxt);
}
}